src/axom/core/Map.hpp

```diff
@@ -383,7 +383,28 @@ class Map
     }
     return ret;
   }
+  
+  /*!
+   * \brief Finds reference to value of key-value pair mapped to supplied pair, and performs insertion if
+   *  item does not already exist in Map instance. For reading, modification, and insertion into Map instance.
+   *
+   * \param [in] key the key used to index into the Map to find/store item
+   *
+   * \note Major deviation from STL unordered_map in that this function is unsafe, due to inability to internally
+   *  call rehash function. User must be careful to ensure integrity of Map with repeated use of this function.
+   *
+   * \return A reference to the value of key-value pair in the Map instance that is associated with supplied Key.
+   *
+   */  
+  T& operator[](Key key)
```

Comment:
Suggestion: Maybe we should only support the `const` version of `operator[]`, at least initially

Reply:
So read-only, thus not performing the auto-insertion step?

Reply:
That's what I was thinking (only a suggestion though).  

Perhaps we can add the non-const version down the line if/when we think of a clever way to do it.

Reply:
While holding to the API of returning a reference, and avoiding exceptions, I am unaware of a more clever way to do this, yes. 

Though, if we wanted to break API and make this return a pointer, nullptr is always available.

Reply:
I think that breaking the API in than manner (i.e. returning a pointer instead of a reference) would be really confusing

Reply:
Definitely would be. Just a thought of how to avoid this without throwing exceptions, which was indicated to me to be a bad thing for when this hits GPU.